src/python/pants/engine/internals/specs_rules.py

```diff
@@ -146,7 +148,9 @@ async def addresses_from_raw_specs_without_file_owners(
         Get(Paths, PathGlobs, validation_globs),
     )
 
-    dirnames = {os.path.dirname(f) for f in build_file_paths.files}
+    # TODO: filter `synthetic_target_paths` based on specs_filter..
```

Comment:
I'd appreciate some input on how to do this properly, as I think the behavior will be odd (or even potentially wrong) unless we do apply some filtering here.

Reply:
Not sure I understand the concern: can you expand on that? You will definitely need to filter, since the entire `filter` subsystem is now applied by `SpecsFilter`, and so things like `--filter-target-type` need to be applied.

Reply:
By not caring at all about synthetic targets in `specs_rules.py` this thread is now moot. But, it doesn't come without a caveat that synthetic targets in directories without any BUILD file in them won't show up in target listings, only when addressed explicitly (by target or file name) (is what I suspect at least, haven't verified yet, and not sure if it matters much but potentially a little bit surprising behaviour)

Reply:
> Not sure I understand the concern: can you expand on that?

Say you run `./pants list x` or `./pants test --changed-since=y` if all the synthesized targets where to show up regardless of what you asked for, that could give all kind of weird issues, if that list was used as input for downstream tasks for instance (odd / wrong depending on use case, I suppose)